### ISSUE TITLE
undefined function ejabberd_http_ws

### ISSUE THREAD

**Issue by vicvolk** [processone/ejabberd#708]
I get this error message in `ejabberd.log` when I try to connect to `ejabberd` from the client code. I think I do everything according to documentation:

   1. To enable `mod_http_ws` I added these two lines to the top of `ejabberd.yml` file:

        websocket_ping_interval: 60

        websocket_timeout: 86400
   2. This block is also present in `ejabberd.yml`:

        listen:
        ...
           -
             port: 5280
             module: ejabberd_http
             request_handlers:
                   "/http-ws": ejabberd_http_ws

   3. And finally in the client code I use this connection string:

         var SocketUrl = 'ws://localhost:5280/http-ws'

         which is later used by `Connection` method of the `Strophe` library:

         conn = new Strophe.Connection(SocketUrl);

Don't you know what is wrong with that?

**Comment by zinid** [comment]
So what error do you get?

**Comment by zinid** [comment]
I'm not sure about original ejabberd.yml, but in the snippet above your indentation looks incorrect : port, module and request_handlers should stand at the same level.

**Comment by vicvolk** [comment]
I have a correct indentation. It is just  a matter of makrdown that formated my message in this way.

**Comment by vicvolk** [comment]
"So what error do you get?" In `ejabberd.yml` I have this error message: `undefined function ejabberd_http_ws `

**Comment by prefiks** [comment]
It should be working with that config fragment, would you be able pastebin your whole config file and log file with that error that you are getting? I will try to see if i get something similar on my own machine, and trace what generates that error.

**Comment by vicvolk** [comment]
I found what was wrong with that. The whole issue was in the default version of `ejabberd` installed by `$ apt-get install ejabberd`. The fact is the version that I first installed lacks `mod_http_ws`. I upgraded to version `15.07`  and now it all works. I think documentation would save me two days if it has just a teeny-weeny comment under `mod_http_ws` - http://docs.ejabberd.im/admin/guide/configuration/#modhttpws - that it is supported only in versions higher than 15.0X
